tests/CNV-72329/implementation-python/test_nad_swap_negative.py

```
@@ -0,0 +1,115 @@
+                    ):
+                        pass
+
+                    LOGGER.info("Verifying migration fails or VM condition set")
+                    # Migration should fail or RestartRequired condition should be set
+                    # VM should remain on source
+                    current_vmi_uid = vm.vmi.instance.metadata.uid
+                    assert current_vmi_uid == original_vmi_uid, "VM should remain on source after failed NAD change"
+
+                    LOGGER.info("Test passed: Non-existent NAD handled correctly")
+
+                except Exception as e:
```

Ruff (0.15.0)

[warning] Do not catch blind exception: `Exception`

(BLE001)